src/SANDECALClustering.cpp

```diff
@@ -0,0 +1,92 @@
+#include "SANDClustering.h"
+
+int clustering(std::string const& input)
+{
+  
+  const char* finname = input.c_str();
+  TFile f(finname, "READ");
+  TTree* t = (TTree*)f.Get("tDigit");
```

Comment:
I'd add a check for the existence of the TTree to prevent a crash when trying to get the entries of the tDigit Tree. Something like that

if (!t) {
std::cerr << "TTree tDigit not found in file " << finname << std::endln;
return -1;
}